### PR TITLE
Coderefactoring

### PR DESCRIPTION
**Pull request by praveensree**

### CODE REVIEW

File: Server/Models/fbContext.cs [hunk 1]
```diff
@@ -20,13 +20,13 @@ public fbContext(DbContextOptions<fbContext> options)
...
         public virtual DbSet<Post> Posts { get; set; }
 
-        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
-        {
-            if (!optionsBuilder.IsConfigured)
-            {
-                optionsBuilder.UseSqlServer("Server=BHAVNAWKS727;Database=fb;Trusted_Connection=True;");
-            }
-        }
+        //protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
Comment (akshayverma91) [inline Server/Models/fbContext.cs:23]: remove commented code

File: Server/Services/PostService.cs [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-using Server.Repository;
+﻿using PostApp.Models;
+using PostApp.Repository;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 
-namespace Server.Services
+namespace PostApp.Services
```
Comment (akshayverma91) [inline Server/Services/PostService.cs:8]: Rename file name to PostService name to SocialPostService.
and its references

File: Server/Repository/PostRepository.cs [hunk 3]
```diff
@@ -1,149 +1,127 @@
...
-
-
-                    return fb.Posts.First(x => x.PostId == id);
-
-                }
-                catch (Exception ex)
-                {
-                    throw ex;
-                }
+                return await context.Posts.FirstAsync(x => x.PostId == id);
```
Comment (akshayverma91) [inline Server/Repository/PostRepository.cs:38]: Can we try to use singleordefault and change to other places if yo get only single record

File: Server/Repository/PostRepository.cs [hunk 4]
```diff
@@ -1,149 +1,127 @@
...
-                    update.PostDescription = post.PostDescription;
-                    update.UpdatedDate = DateTime.Now;
-                    fb.SaveChanges();
-                    return fb.Posts.First(x => x.PostId == id);
-                }
-                catch (Exception ex)
-                {
-                    throw ex;
-                }
+                var update = await context.Posts.FirstAsync(x => x.PostId == id);
```
Comment (akshayverma91) [inline Server/Repository/PostRepository.cs:50]: I think we can use FirstorDefaultAsync

**Comment by akshayverma91** [comment]
Can you refer code standard from client project for error handling in service and controller

File: Server/Models/fbContext.cs [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 #nullable disable
 
-namespace Server.Models
+namespace PostApp.Models
 {
     public partial class fbContext : DbContext
```
Comment (akshayverma91) [inline Server/Models/fbContext.cs:9]: Change to PostAppContext

File: Server/Models/fbContext.cs (hunk 1 above)
Comment (praveensree) [reply Server/Models/fbContext.cs]: Removed the commented code

File: Server/Services/PostService.cs (hunk 2 above)
Comment (praveensree) [reply Server/Services/PostService.cs:8]: changed the Name

File: Server/Repository/PostRepository.cs (hunk 3 above)
Comment (praveensree) [reply Server/Repository/PostRepository.cs]: changed to SingleOrDefault

File: Server/Models/fbContext.cs (hunk 5 above)
Comment (praveensree) [reply Server/Models/fbContext.cs]: changed the name of context to PostAppContext

File: Server/Repository/PostRepository.cs (hunk 4 above)
Comment (praveensree) [reply Server/Repository/PostRepository.cs]: Changed to FirstOrDefaultAsync

**Pull request merged**